src/hps_align/plot/residual.py

```diff
@@ -0,0 +1,174 @@
+import ROOT as r
+from ._plotter import Plotter
+from . import alignment_utils
+from .index_page import htmlWriter
+
+
+def single_residual_plot(p: Plotter, histo_name, title=""):
+    """Plot a single 1D residual plot for the input sensor"""
+
+    histos = [f.Get('res/'+histo_name) for f in p.input_files]
+
+    canv = r.TCanvas("c1", "c1", 2200, 2000)
+
+    if (title == ""):
+        title = histo_name.split("_")[3] + "_" + histo_name.split("_")[5]
+
+    maximum = -1.
+
+    fitList = []
+    plot_properties = []
+
+    # Normalize and get maximum after normalisation
+    for histo in histos:
+        # Normalize the histogram to unity
+        if abs(histo.Integral()) > 1e-8:
+            histo.Scale(1./histo.Integral())
+        # Get the maximum
+        if (histo.GetMaximum() > maximum):
+            maximum = histo.GetMaximum()
+
+    if ("slot") in histo_name:
+        title += "_slot"
+    elif "hole" in histo_name:
+        title += "_hole"
+
+    for ihisto in range(len(histos)):
+        p.set_histo_style(histos[ihisto], ihisto)
+
+        # Fitting
+        fitList.append(alignment_utils.make_fit(
+            histos[ihisto], "singleGausIterative", [], p.colors[ihisto]))
+
+        if (ihisto == 0):
+            histos[ihisto].GetXaxis().SetTitle(title)
+            histos[ihisto].GetXaxis().SetTitleSize(0.05)
+            histos[ihisto].GetXaxis().SetTitleOffset(1.25)
+            histos[ihisto].SetMaximum(maximum*1.5)
+            histos[ihisto].GetYaxis().SetRangeUser(0.0, maximum*1.5)
+            histos[ihisto].Draw("P")
+        else:
+            histos[ihisto].Draw("P SAME")
+
+        fitList[ihisto].Draw("SAME")
+
+        # Save fit properties
+        mu = fitList[ihisto].GetParameter(1)
+        sigma = fitList[ihisto].GetParameter(2)
+
+        plot_properties.append((" #mu=%.3f" % round(mu, 3))
+                               + (" #sigma=%.3f" % round(sigma, 3)))
+
+    leg = p.do_legend(histos, p.legend_names, 3, plot_properties)
+
+    if (leg is not None):
+        leg.Draw()
+
+    text = r.TLatex()
+    text.SetNDC()
+    text.SetTextFont(42)
+    text.SetTextSize(0.04)
+    text.SetTextColor(r.kBlack)
+    text.DrawLatex(0.16, 0.89, '#bf{#it{HPS}} Work In Progress')
+
+    canv.SaveAs(p.outdir + "/" + histo_name + p.oFext)
+
+    if p.do_HTML:
+        hw = htmlWriter(p.outdir)
+        hw.add_images(p.outdir)
+        hw.close_html()
+
+
+@Plotter.user
+def one_dim(p: Plotter):
+    """Plot the 1D residuals for the sensors in the plot listing"""
+
+    for plot in p.plot_list('1D_residual_plots'):
+        single_residual_plot(p, plot)
+
+
+@Plotter.user
+def vs_u(p: Plotter):
```

Comment:
This function is not working for me for whatever reason... res_vs_v is just fine but the res_vs_u output is messed up. It seems like it's not getting all the data from the root file and therefore is not filling the histograms correctly.

Reply:
![uresidual_GBL_vs_u_hit_module_L2b_halfmodule_axial_sensor0_profiled](https://github.com/JeffersonLab/hps-align/assets/69592687/9daf5793-6643-447a-9317-2ca66d8ffc6d)

Reply:
This happens for all layers but the first one

Reply:
https://github.com/JeffersonLab/hps-align/pull/24/files#r1210398808